src/scanoss/agent/src/scanner.c

```diff
@@ -0,0 +1,950 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * src/scanner.c
+ *
+ * A CUSTOMIZED Version of the original SCANOSS  C client for direct file scanning
+ * from the scanoss agent for FOSSOlogy
+ *
+ * Copyright (C) 2018-2020 SCANOSS.COM
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#define _GNU_SOURCE
+#include <ctype.h>
+#include <openssl/md5.h>
+#include <openssl/ssl.h>
```

Comment:
Please consider using [gcrypt](https://www.gnupg.org/documentation/manuals/gcrypt/) instead of openssl since openssl is licensed under Apache v2 from release 3.0.

Please check https://github.com/fossology/fossology/pull/1545/files for usage of hashing functions.

Reply:
Done! Openssl and ssl dependencies were removed. Using GCrypt instead